docs/source/tools/export.rst

```diff
@@ -0,0 +1,164 @@
+.. Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Ledger Export
+#############
+
+Export is currently an :doc:`Early Access Feature in Labs status </support/status-definitions>`.
+
+Introduction
+************
+
+Daml ledger exports read the transaction history or active contract set (ACS)
+from the ledger and write it to disk encoded as a
+:doc:`Daml Script </daml-script/index>` that will reproduce the ledger state
+when executed. This can be useful to migrate the history or state of a ledger
+from one network to another, or to replicate the ledger state locally for
+testing or debugging purposes.
+
+Usage
+*****
+
+The command to generate a Daml ledger export has the following form. ::
+
+  daml ledger export <format> <options>
+
+Right now Daml script, specified as ``script``, is the only supported export
+format. You can get an overview of the available command-line options using the
+``--help`` flag as follows. ::
+
+  daml ledger export script --help
+
+A full example invocation looks like this: ::
+
+  daml ledger export script --host localhost --port 6865 --party Alice --party Bob --output ../out --sdk-version 0.0.0
+
+The flags ``--host`` and ``--port`` define the Daml ledger to connect to. You
+can omit these flags if you are invoking the command from within a Daml project
+with a running ledger, e.g. with a running ``daml start``.
+
+The ``--party`` flags define which contracts will be included in the export. In
+the above example only contracts visible to the parties Alice and Bob will be
+included in the export. Lack of visibility of certain events may cause exports to fail, see :ref:`caveats <export-caveats>`.
+
+The ``--output`` flag defines the directory prefix under which to generate the
+Daml project that contains the Daml script that represents the ledger export.
+The flag ``--sdk-version`` defines which Daml Connect version to configure in
+the generated ``daml.yaml`` configuration file.
+
+By default an export will reproduce all transactions in the ledger history. The
+:ref:`ledger offsets <export-ledger-offsets>` section describes how to change
+this behavior.
+
+Output
+******
+
+The generated Daml code contains the following top-level definitions:
+
+``data Parties``
```

Comment:
Side note: There is no nice mapping from old party ids to new party ids atm. One solution for that would be to accept a TextMap instead of a record as an argument with the keys being the old party ids.

Reply:
Indeed, I've raised a ticket for this: https://github.com/digital-asset/daml/issues/9453